[CCI] Fix EUI/OUI type errors

Nicksqain opened the pull request:
### Description

- [x] Update  imports
- [x] fix EuiIconType type error
- [x] fix query type error in EuiSearchBar

### Issues Resolved

- closes #3028 

### Check List

- [x] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

Nicksqain commented:
I initialised the PR to keep working on it! 
I want to know if the imports will work correctly, would you be kind enough to give feedback?

Nicksqain commented:
@joshuarrrr @ashwin-pc 
I fixed the non-working imports that were declared in elastic ui, accidentally came across a folder with findTestSubject, which is identical and does not cause errors.
But what did I do wrong? I can't run snapshot
![изображение](https://user-images.githubusercontent.com/67852536/230641060-3499e365-ff3a-4dcf-ac10-f0035b6b544e.png)

joshuarrrr commented:
@Nicksqain It's not your fault or problem - it appears there's a problem with the windows snapshot images.

joshuarrrr commented:
@Nicksqain The snapshot should be working again, I restarted the test automation.

joshuarrrr commented:
When you get a chance, can you add an entry to the CHANGELOG.md under the "Refactoring" section?

manasvinibs commented:
@Nicksqain Were you able to get the snapshot running? Also, can you add it to description on how did you test these changes?

Nicksqain commented:
@manasvinibs 
Thank you for your concern! I was able to run the snapshot! Working on that issue at the moment. It's not very clear to me, but it's interesting to figure it out.

Tests? If you mean imports of tests, I've been testing them like this
yarn test:jest src/plugins/advanced_settings/public/management_app/components/field/field.test.tsx

![изображение](https://user-images.githubusercontent.com/67852536/232573703-abb337ea-93b2-4242-8e01-f7eca7690522.png)

joshuarrrr commented:
@Nicksqain Also please update the changelog, when you get the chance.

ashwin-pc commented on src/plugins/advanced_settings/public/management_app/components/field/field.test.tsx:
```diff
@@ -36,7 +36,7 @@ import { FieldSetting } from '../../types';
 import { notificationServiceMock, docLinksServiceMock } from '../../../../../../core/public/mocks';
 
-import { findTestSubject } from '@elastic/eui/lib/test';
+import { findTestSubject } from 'test_utils/helpers';
```
Nice find! yep this function is equivalent to the eui one.

ashwin-pc reviewed (commented):
@Nicksqain thanks for updating the changelog. I see that you have 3 tasks listed in this PR but have only completed one of them. Are you planning on completing all three in the same PR?

Nicksqain commented:
@ashwin-pc I have updated the tasks. I have added changes to the entire PR. Please check, because I'm not sure if it's the way to use Query import. But this import is declared in eui

ashwin-pc commented on src/plugins/advanced_settings/public/management_app/components/search/search.tsx:
```diff
@@ -30,8 +30,8 @@
-import { EuiSearchBar, EuiFormErrorText, Query } from '@elastic/eui';
-
+import { EuiSearchBar, EuiFormErrorText } from '@elastic/eui';
+import { Query } from '@opensearch-project/oui/src/eui_components/search_bar/query';
```
This change add a type error in the parent component that uses `Search`. I suggest that you instead add a `ts-ignore` comment to the ts error on line 114. e.g.

```ts
        {/* @ts-ignore The Query types that typescript complains about here are identical and is a false flag. Once OUI migration is complete, this ignore can be removed */}
        <EuiSearchBar box={box} filters={filters} onChange={this.onChange} query={query} />

```

Nicksqain replied:
@ashwin-pc  That is, do not need to update the import? Will the Query error be fixed in the future?

ashwin-pc replied:
Yeah, we will replace all eui references to oui. That should fix it :)

ashwin-pc reviewed (commented):
Thanks for updating the PR @Nicksqain. The rest of the changes look good to me, just one change needed

BSFishy reviewed (approved):
+1 for removing references to `@elastic/eui/lib/test`. This will make migrations from OUI v1 to v2 much easier

ashwin-pc reviewed (approved):
Nice to see this finally over the finish line. @Nicksqain Tanks for sticking with it  🎉

manasvinibs reviewed (approved):
LGTM!
